hilti/toolchain/src/compiler/visitors/printer.cc

```diff
@@ -982,7 +982,22 @@ struct Visitor : visitor::PreOrder<void, Visitor> {
         if ( n.isWildcard() )
             out << const_(n) << "tuple<*>";
         else {
-            out << const_(n) << "tuple<" << std::make_pair(n.types(), ", ") << ">";
+            out << const_(n) << "tuple<";
+
+            bool first = true;
+            for ( const auto& [id, ty] : n.elements() ) {
+                if ( ! first )
+                    out << ", ";
+
+                if ( id )
+                    out << id << ": " << ty;
+                else
+                    out << ty;
+
+                first = false;
+            }
```

Comment:
This looks equivalent to a `transform` and a `join`,

```suggestion
            out << util::join(util::transform(n.elements(),
                                              [](auto&& element) {
                                                  auto&& [id, ty] = element;

                                                  if ( id )
                                                      return fmt("%s: %s", id, ty);
                                                  else
                                                      return fmt("%s", ty);
                                              }),
                              ",");

```

This requires an `#include <hilti/base/util.h>`.